parsers/CY.py

```diff
@@ -115,10 +119,8 @@ def fetch_production(
     """Requests the last known production mix (in MW) of a given country."""
     assert zone_key == "CY"
 
-    target_datetime = target_datetime or datetime.utcnow()
-
     parser = CyprusParser(session or Session(), logger)
-    if isinstance(target_datetime, datetime):
+    if isinstance(target_datetime, Optional[datetime]):
         return parser.fetch_production(target_datetime)
```

Comment:
This won't work. The isinstance method can not be used with a optional type (at least not in python 3.8). But I don't think you need it at all with the other changes you have made.
```suggestion
    return parser.fetch_production(target_datetime)
```